appcues/src/main/java/com/appcues/debugger/screencapture/AndroidTargetingStrategy.kt

```diff
@@ -176,6 +193,64 @@ private fun View.asCaptureView(screenBounds: Rect): ViewElement? {
     }
 }
 
+private suspend fun WebView.children(positionAdjustment: Rect): List<ViewElement> {
+    val js = """
+        [...document.querySelectorAll('[id], [appcues-id]')].reduce((result, el) => {
+            const { x, y, width, height } = el.getBoundingClientRect();
+            const tag = el.id ? `#${'$'}{el.id}` : null;
+            const appcuesID = el.getAttribute('appcues-id')
+            if (height !== 0 && width !== 0) {
+                result.push({
+                    x,
+                    y,
+                    width,
+                    height,
+                    tag,
+                    appcuesID
+                });
+            }
+            return result;
+        }, []);
+        """
+
+    val result = evaluateJavascript(js)
+    return result.map { el ->
+        val x = (el["x"] as Double).toInt()
+        val y = (el["y"] as Double).toInt()
+        val width = (el["width"] as Double).toInt()
+        val height = (el["height"] as Double).toInt()
+        ViewElement(
+            x = positionAdjustment.left + x,
+            y = positionAdjustment.top + y,
+            width = width,
+            height = height,
+            displayName = null,
+            selector = AndroidViewSelector(
+                properties = mapOf(
+                    SELECTOR_APPCUES_ID to (el["appcuesID"] as String?),
+                    SELECTOR_TAG to (el["tag"] as String?)
+                ).filterValues { it != null },
+            ),
+            type = "HTMLNode",
+            children = null,
+        )
+    }
+}
+
+private suspend fun WebView.evaluateJavascript(script: String): List<Map<String, Any>> {
+    val completion = CompletableDeferred<List<Map<String, Any>>>()
+    evaluateJavascript(script) { result ->
+        val listType: Type = Types.newParameterizedType(
+            List::class.java,
+            Types.newParameterizedType(Map::class.java, String::class.java, Any::class.java)
+        )
+        val adapter = MoshiConfiguration.moshi.adapter<List<Map<String, Any>>>(listType)
+        val parsed: List<Map<String, Any>> = adapter.fromJson(result)!!
```

Comment:
took a quick peek to see if there was some simpler way to express this for the decoding, but no luck! I guess this is the way.

Reply:
Reminds me I was going to revisit the non-null assertion. I think this would be preferable?
```kt
val parsed: List<Map<String, Any>> = adapter.fromJson(result) ?: emptyList()
```

Reply:
yeah agreed that would be a safer approach